OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -606,43 +606,156 @@ static class DMOpcodeHandlers {
             DreamValue second = state.Pop();
             DreamValue first = state.Pop();
 
+            switch (first.Type) {
+                case DreamValue.DreamValueType.Float when second.Type == DreamValue.DreamValueType.Float:
+                    state.Push(new DreamValue(first.GetValueAsInteger() << second.GetValueAsInteger()));
+                    break;
+                default:
+                    throw new Exception("Invalid bit shift left operation on " + first + " and " + second);
+            }
+
+            return null;
+        }
+
+        public static ProcStatus? Output(DMProcState state)
+        {
+            var reference = state.ReadReference();
+
+            DreamValue left;
+            DreamValue idx = DreamValue.Null;
+
+            switch (reference.RefType)
+            {
+                case DMReference.Type.ListIndex:
+                    idx = state.GetReferenceValue(reference, true);
```

Comment:
`GetReferenceValue()` is attempting to index a savefile and failing because it's not a list or string.
```suggestion
                    idx = state.Pop();
```